PokemonRampUp/ViewController/LandingViewController.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  LandingViewController.swift
+//  PokemonRampUp
+//
+//  Created by Thushen Mohanarajah on 2023-02-09.
+//
+import UIKit
+
+class LandingViewController: UIViewController {
+
+    @IBOutlet weak var tableView: UITableView!
+    @IBOutlet weak var loadingView: UIView!
+    
+    private let amountOfPokemon = 5
+    private var pokemons: [Pokemon]?
+    private let webServices = WebServices()
+    private let refreshControl = UIRefreshControl()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        tableView.delegate = self
+        tableView.dataSource = self
+        refreshControl.addTarget(self, action: #selector(refetchPokemon), for: .valueChanged)
+        tableView.refreshControl = refreshControl
+        pokemons = []
+        fetchPokemon()
+    }
+    
+    private func fetchPokemon() {
+        Task {
+            do {
+                var count = 0;
+                while (count < amountOfPokemon){
+                    self.pokemons?.append(try await webServices.fetchRandomPokemon())
+                    count += 1
+                }
+                self.refreshControl.endRefreshing()
+                tableView.reloadData()
+                loadingView.isHidden = true
+            }
+            catch {
+                loadingView.isHidden = true
+                let alert = UIAlertController(title: "Error Loading Pokémon", message: "Please try again or exist app", preferredStyle: .alert)
```

Comment:
can you create a factory for the alertController? in order to avoid repeating the alertcontroller all over your codebase

Reply:
I added a factory👍